internal/network/manifests_generator.go

```diff
@@ -103,46 +169,118 @@ func createChronyManifestContent(c *common.Cluster, role models.HostRole) (strin
 }
 
-func (m *ManifestsGenerator) AddChronyManifest(ctx context.Context, log logrus.FieldLogger, c *common.Cluster) error {
+func (m *ManifestsGenerator) AddChronyManifest(ctx context.Context, log logrus.FieldLogger, cluster *common.Cluster) error {
 	for _, role := range []models.HostRole{models.HostRoleMaster, models.HostRoleWorker} {
-		content, err := createChronyManifestContent(c, role)
+		content, err := createChronyManifestContent(cluster, role, log)
 
 		if err != nil {
-			return errors.Wrapf(err, "Failed to create chrony manifest content for role %s cluster id %s", role, *c.ID)
+			return errors.Wrapf(err, "Failed to create chrony manifest content for role %s cluster id %s", role, *cluster.ID)
 		}
 
 		chronyManifestFileName := fmt.Sprintf("%ss-chrony-configuration.yaml", string(role))
-		folder := models.ManifestFolderOpenshift
-		base64Content := base64.StdEncoding.EncodeToString([]byte(content))
-
-		response := m.manifestsApi.CreateClusterManifest(ctx, operations.CreateClusterManifestParams{
-			ClusterID: *c.ID,
-			CreateManifestParams: &models.CreateManifestParams{
-				Content:  &base64Content,
-				FileName: &chronyManifestFileName,
-				Folder:   &folder,
-			},
-		})
-
-		if _, ok := response.(*operations.CreateClusterManifestCreated); !ok {
-			if apiErr, ok := response.(*common.ApiErrorResponse); ok {
-				return errors.Wrapf(apiErr, "Failed to create manifest %s", chronyManifestFileName)
-			}
-
-			return errors.Errorf("Failed to create manifest %s", chronyManifestFileName)
+		err = m.createManifests(ctx, cluster, chronyManifestFileName, content)
+		if err != nil {
+			return err
 		}
 	}
 
 	return nil
 }
+
+func (m *ManifestsGenerator) createManifests(ctx context.Context, cluster *common.Cluster, filename string, content []byte) error {
+	// all relevant logs of creating manifest weill be inside CreateClusterManifest
+	response := m.manifestsApi.CreateClusterManifest(ctx, operations.CreateClusterManifestParams{
+		ClusterID: *cluster.ID,
+		CreateManifestParams: &models.CreateManifestParams{
+			Content:  swag.String(base64.StdEncoding.EncodeToString(content)),
+			FileName: &filename,
+			Folder:   swag.String(models.ManifestFolderOpenshift),
+		},
+	})
+
+	if _, ok := response.(*operations.CreateClusterManifestCreated); !ok {
+		if apiErr, ok := response.(*common.ApiErrorResponse); ok {
+			return errors.Wrapf(apiErr, "Failed to create manifest %s", filename)
+		}
+		return errors.Errorf("Failed to create manifest %s", filename)
+	}
+	return nil
+}
+
+func (m *ManifestsGenerator) AddDnsmasqForSingleNode(ctx context.Context, log logrus.FieldLogger, cluster *common.Cluster) error {
+	filename := "dnsmasq-bootstrap-in-place.yaml"
+
+	content, err := createDnsmasqForSingleNode(log, cluster)
+	if err != nil {
+		return err
+	}
+
+	return m.createManifests(ctx, cluster, filename, content)
+}
+
+func createDnsmasqForSingleNode(log logrus.FieldLogger, cluster *common.Cluster) ([]byte, error) {
```

Comment:
Consider renaming
```suggestion
func GetDnsmasqForSingleNodeManifestContent(log logrus.FieldLogger, cluster *common.Cluster) ([]byte, error) {
```

Reply:
this is private function, was added only for unitest, to be able to test creation itself.